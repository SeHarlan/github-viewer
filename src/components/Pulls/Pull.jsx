import React from 'react';
import PropTypes from 'prop-types';

const Pull = ({ pullBody, pullNumber }) => (
  <em>{pullNumber}: {pullBody}</em>
);

Pull.propTypes = {
  pullBody: PropTypes.string.isRequired,
  pullNumber: PropTypes.number.isRequired
};

export default Pull;
