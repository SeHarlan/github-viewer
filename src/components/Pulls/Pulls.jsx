import React from 'react';
import PropTypes from 'prop-types';
import Pull from './Pull.jsx';

const Pulls = ({ pulls, closedOrOpen }) => {
  //state can be closed or open
  const pullElements = pulls
    .filter(pull => {
      if(closedOrOpen) return (pull.state === closedOrOpen);
      return true;
    })
    .map(pull => <Pull key={pull.id} pullNumber={pull.number} pullBody={pull.body}/>);
  return <ul>{pullElements}</ul>;
};

Pulls.propTypes = {
  pulls: PropTypes.array.isRequired,
  closedOrOpen: PropTypes.string
};

export default Pulls;
