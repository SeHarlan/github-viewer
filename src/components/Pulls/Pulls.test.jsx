import React from 'react';
import { shallow } from 'enzyme';
import Pulls from './Pulls.jsx';

describe('Pulls component', () => {
  it('renders Pulls', () => {
    const wrapper = shallow(<Pulls pulls={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
