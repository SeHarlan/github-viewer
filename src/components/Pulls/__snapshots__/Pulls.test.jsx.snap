// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pulls component renders Pulls 1`] = `<ul />`;
